src/message_bus.zig

```diff
@@ -235,8 +235,9 @@ fn MessageBusImpl(comptime process_type: ProcessType) type {
             for (self.connections) |*connection| {
                 if (connection.state == .idle) {
                     assert(connection.peer == .none);
-                    // This function immediately adds the connection to Self.replicas.
+                    // This function immediately adds the connection to self.replicas.
                     connection.connect_to_replica(self, replica);
+                    assert(connection.peer.replica == replica);
```

Comment:
we could also
```zig
assert(self.replicas[replica] == connection);
```

Reply:
Thanks! That's a good idea and I was wondering why we didn't do that from the beginning instead of the comment. I took a closer look at `connect_to_replica()` and it turns out we have an early return in there if a socket cannot be obtained, so that we cannot actually assert anything here after the function call.